client.go

```diff
@@ -182,6 +223,19 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		req.Method, req.URL, c.RetryMax+1)
 }
 
+// Try to read the response body so we can reuse this connection.
+func (c *Client) drainBody(body io.ReadCloser) {
+	defer body.Close()
+	_, err := io.Copy(ioutil.Discard, &io.LimitedReader{R: body, N: respReadLimit})
+	if err != nil {
+		// don't log this if the body was closed by the ResponseLogHook or CheckRetry
+		if strings.Contains(err.Error(), "read on closed response body") {
```

Comment:
Can we just close the response body automatically near line 198 instead, and update the documentation such that the body close is handled for the client? In that case we know we are not retrying and that a custom error was returned. I always like to avoid switching on error strings when possible.